### ISSUE TITLE
Read Timeout and Open Timeout is not working (under savon)

### ISSUE THREAD

**Issue by ehutzelman** [savonrb/httpi#192]
I'm not sure if this is only when using httpi with savon, but after having problems I discovered this related issue under savon: https://github.com/savonrb/savon/issues/797

I wanted to open it up here to see if there is something that needs to change in httpi to correct this. As the other issue states, you can rollback to httpi version '2.3.0' to get timeouts working again with savon.

**Comment by rogerleite** [comment]
Hi @ehutzelman 

Open and read timeout are set by adapters. The only adapters that not set these options are [http](https://github.com/savonrb/httpi/blob/master/lib/httpi/adapter/http.rb) and rack, all others set these options.

* [curb](https://github.com/savonrb/httpi/blob/master/lib/httpi/adapter/curb.rb#L75)
* [em_http](https://github.com/savonrb/httpi/blob/master/lib/httpi/adapter/em_http.rb#L72)
* [excon](https://github.com/savonrb/httpi/blob/master/lib/httpi/adapter/excon.rb#L59)
* [httpclient](https://github.com/savonrb/httpi/blob/master/lib/httpi/adapter/httpclient.rb#L46)
* [net_https](https://github.com/savonrb/httpi/blob/master/lib/httpi/adapter/net_http.rb#L156)
* [net_http_persistent](https://github.com/savonrb/httpi/blob/master/lib/httpi/adapter/net_http_persistent.rb#L33)

To help identify this bug, it's important to describe what adapter (and version) you're using.

If you don't know what adapter you're using, remember that httpi try to pick one using [LOAD_ORDER](https://github.com/savonrb/httpi/blob/master/lib/httpi/adapter.rb#L16) constant.

Hope that helps!
Regards.

**Comment by ehutzelman** [comment]
Thanks for the quick response @rogerleite, and now that I tested again this may be a non-issue. I don't know if the upgrade to ruby 2.5.1 had an effect or possibly another gem, but I just tested again with httpi 2.4.3 and the timeout setting is being honored as expected. For the record, I'm using `HTTPI.adapter = :net_http` for the adapter. Sorry for the false alarm, it seems to have resolved itself.
